analyzer/decrypt_crypto.cc

```diff
@@ -136,7 +136,7 @@ std::vector<uint8_t> hkdf_expand(size_t out_len,
 /*
 Removes the header protection from the INITIAL packet and returns a DecryptionInformation struct that is partially filled
 */
-DecryptionInformation remove_header_protection(std::vector<uint8_t> client_hp, uint8_t encrypted_offset, std::vector<uint8_t> encrypted_packet)
+DecryptionInformation remove_header_protection(std::vector<uint8_t> client_hp, uint64_t encrypted_offset, std::vector<uint8_t> encrypted_packet)
```

Comment:
Now that we pass an `uint64_t` I am less concerned about potential overflows in the unchecked additions below.

Just FYI, if you would like to opt into safe integers in the future or even now, just use `hilti::rt::integer::safe<uint8_t>` or similar as an arg. These types add safety checks to operations, but have the same memory footprint.

Also since you are touching this: Let's add `#include <cstdint>`.

Reply:
Fairly sure the `uint8_t` has not been correct because offsets could be larger than 255.